jsplot/src/main/java/tech/tablesaw/plotly/components/Component.java

```diff
@@ -12,11 +15,33 @@
 public abstract class Component {
 
   protected final PebbleEngine engine = TemplateUtils.getNewEngine();
+  protected final ObjectMapper mapper = new ObjectMapper();
 
+  {
+    mapper.enable(SerializationFeature.INDENT_OUTPUT);
+    mapper.setSerializationInclusion(Include.NON_NULL);
+    mapper.enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS);
+  }
+
+  @Deprecated
   public abstract String asJavascript();
 
+  @Deprecated
```

Comment:
You could probably just remove these methods rather than deprecating. I doubt many people are extending these classes vs using them directly

Reply:
Right now we have components generating JS with PebbleEngine and Jackson. So it makes sense to deprecate this until PebbleEngine can be removed entirely.